packages/grafana-data/src/field/fieldOverrides.ts

```diff
@@ -436,33 +469,90 @@ export const getLinksSupplier =
     fieldScopedVars: ScopedVars,
     replaceVariables: InterpolateFunction,
     timeZone?: TimeZone,
-    dataLinkPostProcessor?: DataLinkPostProcessor
+    dataLinkPostProcessor?: DataLinkPostProcessor,
+    enhancedReplaceVariables?: EnhancedInterpolateFunction
   ) =>
   (config: ValueLinkConfig): Array<LinkModel<Field>> => {
     if (!field.config.links || field.config.links.length === 0) {
       return [];
     }
-
     const linkModels = field.config.links.map((link: DataLink) => {
       const dataContext: DataContextScopedVar = getFieldDataContextClone(frame, field, fieldScopedVars);
+
+      // if we have a row index, expose every field on row as a variable
+      let rowVars: ScopedVars = {};
+      if (config.valueRowIndex !== undefined) {
+        let fieldDisplayValuesProxy: Record<string, DisplayValue> | undefined = getFieldDisplayValuesProxy({
+          frame,
+          rowIndex: config.valueRowIndex,
+        });
+
+        // add all fields in frame as variables
+        frame.fields.forEach((f) => {
+          if (fieldDisplayValuesProxy && fieldDisplayValuesProxy[f.name]) {
+            rowVars[f.name] = {
+              value: fieldDisplayValuesProxy[f.name],
+            };
+          }
+        });
+      }
+
+      // expose variables from transformations
+      let transformationScopedVars = {};
+      if (link.meta?.transformations) {
+        link.meta?.transformations.forEach((transformation) => {
+          let fieldValue;
+          if (config.valueRowIndex !== undefined) {
+            if (transformation.field) {
+              const transformField = frame?.fields.find((field) => field.name === transformation.field);
+              fieldValue = transformField?.values[config.valueRowIndex];
+            } else {
+              fieldValue = field.values[config.valueRowIndex];
+            }
+          } else if (config.calculatedValue) {
+            fieldValue = config.calculatedValue;
+          }
```

Comment:
Unsure what is going on here and how this is needed. 

 what's the problem/scenario (unit tests for applyFieldOverrides or getLinkSupplier would help.